homework/unique_ptr/unique_ptr.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+template<typename T>
+class unique_ptr {
+public:
+    unique_ptr() = default;
+    unique_ptr(T* pointer) : pointer_(pointer) {}
+    unique_ptr(unique_ptr& ptr);
+    unique_ptr(unique_ptr&& ptr);
+    ~unique_ptr();
+    T& operator*() const { return *pointer_; }
+    T* operator->() const { return pointer_; }
+    T* get() const { return pointer_; }
+    T* release();
+    void reset(T* new_ptr);
+private:
+    T* pointer_ = nullptr;
+};
+
+template<typename T>
+unique_ptr<T>::~unique_ptr() {
+    if(pointer_ != nullptr) {
+        delete pointer_;
+        pointer_ = nullptr;
+    }
```

Comment:
just
```suggestion
    delete pointer_;
```
It's safe to delete nullptr.